### ISSUE TITLE
[1.14.4-1.0.0] does not generate a config

### ISSUE THREAD

**Issue by kreezxil** [vincentmet/CustomQuests#3]
at least it doesn't crash the server now.

server config/ folder listing: https://gist.github.com/deeb13a4b6ae3e8012ea96e9cc8f45be

client config/ folder listing: https://gist.github.com/739f63f121821fb4167982a48644f972

**Comment by vincentmetevelis** [comment]
What actions did you perform before looking at the configs?!
Because the the files should generate/load whenever the world is loaded (subscribing to the 'net.minecraftforge.event.world.WorldEvent.Load' class).
For servers that is a couple seconds before you're able to join the server from the multiplayer menu; and for singleplayer that is whenever you create/open a new world save.
Hope this helps! and let me know if it still not works after this

**Comment by kreezxil** [comment]
I installed your updated mod and restarted my server. no configs were generated not even in the world/serverconfig/ folder.

Looking at your code, you config  handler isn't even registered in the base.class which to me explains why it doesn't generate configs. The @Mod.EventSubscriber annotation is no longer enough. I ran into this issue while creating "Once-Upon-A-Stroll".

https://github.com/kreezxil/Once-Upon-a-Stroll/tree/1.14.4/src/main/java/com/kreezcraft/onceuponastroll

**Comment by kreezxil** [comment]
granted mine isn't registered either, but I do call its init.

**Comment by vincentmetevelis** [comment]
I don't register a config handler because I don't have one. In the constructor I'm registering my event handler, in the event handler there's a WorldEvent.Load event, where load the json via the JsonHandler and StructureHandler

**Comment by kreezxil** [comment]
so what you're telling me is that when you load it in your test pack, that
you're getting the config file? and the config file is appearing the in the
config folder on the client or the server? are you even testing it on a
server build?

**Comment by vincentmetevelis** [comment]
exactly, i just tested it again on a forge 28.1.106 server, Questbook and Quests file generate fin in the config folder, and so does the progress file in the world folder

**Comment by kreezxil** [comment]
I'll create a test pack with just this mod and see what the result is.

**Comment by kreezxil** [comment]
I don't see how a pre-existing world/pack installation would prevent it
from generating the config.

**Comment by kreezxil** [comment]
I think the forge version might be at fault here. When I opened this I using FORGE 28.1.109 and now I'm on 28.1.111 and still no config.

Which forge version are you using specifically? Also I can't roll down to it w/e it is as it'll break half the mods in my packs. It should be noted that I have at least 6 1.14.4 packs.

**Comment by kreezxil** [comment]
ok I see you are on

```json
dependencies {
    minecraft 'net.minecraftforge:forge:1.14.4-28.1.106'
	//put JEI here as soft dependency
}
```

I'll make a test pack on that version and then another on the higher version and see if it is not working because of the higher version.

if it continues to work on the higher version, I'll then proceed to see which mod is causing the conflict. if that be the case we're well on our way to creating your incompatibility list.

**Comment by kreezxil** [comment]
ok, just your mod on forge build 106 in a custom server environment, it's not working, it didn't create a config.

server latest.log: https://gist.github.com/f7b048900d7f3d12982e57161b10b982

ok, i found it. you've made hardcoded windows paths so it appears on the top level and not even in the server folder. You're incompatible with linux and server paths. In your code switch from using "\" for pathing to "/" and that will fix this problem for 106.

```
config
config\QuestsBook.json
config\Quests.json
    
```

will now test with a higher forge build, just your mod.

**Comment by kreezxil** [comment]
btw, java treats "/" as universal for paths, and will make the switch to windows behind the scenes for you.

https://stackoverflow.com/questions/19762169/forward-slash-or-backslash

look at the answer with 25 votes (top answer)

**Comment by kreezxil** [comment]
ok, good news it built on the higher forge version too 28.1.111

There is a mod conflict that I will need to find now. Will have this data shortly, testing with a small pack.

**Comment by kreezxil** [comment]
but yes it built wrong, ie i'm testing on a linux server which most of the commerical minecraft hosters use.

**Comment by kreezxil** [comment]
splitting the mods apart to find which one was conflicting to stop config generation didn't work because when I did the issue went away.

ok, this is really odd, so i put all the mods back on and the config generated, granted in the wrong spot, but it generated. I did delete the world a few times.

I wonder, if that's the issue, because you said you got it hooked into world load instead of in the standard places like my mod uses. Meaning that quite possibly if world already exists and this mod gets added, no config will generate. It's a thought. Try placing the config.init in another place like pre-init.

I'm done debugging this for now.

**Comment by vincentmetevelis** [comment]
Oke, ill try using the correct separator, and test it in a linux VM.

The WorldEvent.Load runs everytime a world is loaded, not just when the world is generated.
I also tried putting it in the constructor/onCommonSetup event, but that way i wouldn't have a way to get the world save folder (where the progress is saved)

**Comment by vincentmetevelis** [comment]
and if you want me to on discord (usually respond faster there), here's my tag: vincentmet#1337

**Comment by kreezxil** [comment]
if you want to keep your issue tracker clear sure.

**Comment by vincentmet** [comment]
this is fixed in update 1.0.1 with the #4 merge to master
